tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -393,27 +399,36 @@ export class ScalarCardContainer implements CardRenderer, OnInit, OnDestroy {
       shareReplay(1)
     );
 
-    combineLatest([partitionedSeries$, this.lineChartZoom$]).subscribe(
-      ([series, viewPort]) => {
-        const allPoints = series
-          .map(({points}) => points.map(({x}) => x))
-          .flat();
-        const min =
-          allPoints.length === 0 ? DEFAULT_MIN : Math.min(...allPoints);
-        const max =
-          allPoints.length === 0 ? DEFAULT_MAX : Math.max(...allPoints);
-        const minStep = Math.max(min, viewPort.minStep);
-        const maxStep = Math.min(max, viewPort.maxStep);
-
-        this.minMaxSteps$.next({minStep, maxStep});
+    combineLatest([partitionedSeries$, this.lineChartZoom$])
+      .pipe(
+        withLatestFrom(this.store.select(getMetricsCardMinMax, this.cardId)),
+        map(([[series, viewPort], minMax]) => {
+          if (!isMinMaxStepValid(viewPort) && isMinMaxStepValid(minMax)) {
```

Comment:
Can you describe a situation where the viewport will not be valid?

Reply:
After any kind of zoom action the viewport will be valid.

Reply:
Exactly, when will it not be valid?

Reply:
ok I misread the question in the opposite way. It will be invalid on time seres data loading.

Reply:
to be specific, when `partitionedSeries$` emit values